service/game.ts

```diff
@@ -0,0 +1,60 @@
+import { match, P } from "ts-pattern";
+import { read, Move } from "../model/move";
+import { lastGame, allGames } from "../sql/db";
+import { Result, resultArray, results } from "../model/result";
+import { logRes } from "../sql/db";
+
+export function generateComputerMove(): Move {
+  return read(String(Math.round(Math.random() * 2)));
+}
+export function playLogic(userMove: Move, computerMove: Move): string {
+  return match([userMove, computerMove])
+    .with(
+      ["Rock", "Scissors"],
+      ["Scissors", "Paper"],
+      ["Paper", "Rock"],
+      () => {
+        logRes("WIN");
```

Comment:
Credo sia un problema chiamare una funzione `async` senza farne `await` perché non stai aspettando la Promise da nessuna parte, quindi ti perdi un possibile errore (o meglio, credo Node lo gestisca terminando l'applicazione ma non sono sicuro). Tendenzialmente sarebbe meglio farne `await` (rendendo `async` anche questa funzione) così se c'è un errore nell'eseguire la query puoi dare una risposta di errore al chiamante dell'API, cosa che appunto adesso mi sa che non succede anche se non sono sicurissimo.

Per fare `await` magari è meglio se il match si limita a restituire un risultato `WIN/LOSE/DRAW` di cui poi fai `logRes`, quindi fai un unico `logRes` dopo il `match` invece di farlo in ogni ramo del `match`.

Reply:
Yep, mi torna.